### ISSUE TITLE
[S027] Compute stable next-token cross-entropy loss

### ISSUE THREAD

**Issue by 25d-228** [25d-228/gpt-2#29]
**Prerequisites:** S010, S017; #1 applies. **Timebox:** 30–45 minutes.

## Learn and implement
Create `src/loss.c`, declarations in `include/ops.h`, and `tests/test_loss.c`. For each logit row z and target y, compute `max(z)+log(sum(exp(z-max(z))))-z[y]`; average over all N=B*T predicted tokens. Accumulate the scalar loss in double precision from float logits. Validate target range, finite logits, and nonempty rows. Do not compute `-log(softmax[y])` with an arbitrary probability clamp; that hides extreme-logit errors.

## Explicit tests
Uniform logits over V=4 must give `log(4)` for any legal target. `[0,log(2),log(3)]` targeting index 2 gives `log(2)`. `[1000,0]` targeting index 1 gives approximately 1000, finite. Adding a constant to a whole row must not change loss. Two rows with different losses must average correctly; duplicating all rows must preserve mean loss. Reject invalid targets and NaN/Inf.
```sh
make test TEST=loss
make sanitize TEST=loss
make test
```
All pass with #1 tolerances. Temporarily omit division by N and demonstrate the multirow test fails.

## Done / teach-back
Explain negative log-likelihood, the direction in which better predictions move loss, and mean versus summed loss. No backward or perplexity yet.
